### PR TITLE
SQL-based internal models

### PR DESCRIPTION
**Pull request by chunningham**
# Description

This PR implements an SQL entity based model system to represent the internal state of an orbit peer using sea-orm. It adds a new crate `kepler-core` to the workspace. This new crate:
- defines sql tables for storing authorisation status
- implements authorisation policy enforcement when delegating, invoking or revoking
  - see `delegation`, `invocation` and `revocation` within `kepler_core::models` for policy enforcement and processing implementations
- defines blob storage traits which are broken out from the previous `ImmutableStore`
- wraps a db connection to provide a processing api
- exports `OrbitDatabase` as a wrapper for all functionality

In the existing crate:
- the integration into rocket has been greatly simplified by implementing all logic within `kepler-core`. 
- `store`, `capabilities`, `cas` and other modules have been removed, being superseded by `kepler-core`

# Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Diligence Checklist

- [x] This change requires a documentation update
- [ ] I have included unit tests
- [ ] I have updated and/or included new integration tests
- [ ] I have updated and/or included new end-to-end tests
- [ ] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by sbihel — APPROVED** [review]
This is obviously too much code to review for my limited capabilities (pun intended) -- I've done a code walk through the invocation endpoint, it looks good. The amount of generics and abbreviations makes it hard to follow sometimes. 

The staging storage for hashing will probably become problematic at scale, but you're already aware of it and I don't have any idea to offer.

File: Cargo.toml [hunk 1]
```diff
@@ -62,7 +64,8 @@ path = "lib/"
 members = [
     "lib",
     "sdk-wasm",
-    "sdk"
+    "sdk",
+    "kepler-core"
```
[the remark this answers is not in the document]
Comment (chunningham) [reply Cargo.toml:68]: I did try that, but cargo forbids naming crates after the standard rust ones, e.g. `core` and `std`

File: src/storage/s3.rs [hunk 2]
```diff
@@ -1,45 +1,33 @@
...
-use tokio_util::compat::TokioAsyncReadCompatExt;
 
-use crate::{
-    orbit::ProviderUtils,
-    storage::{utils::copy_in, Content, ImmutableStore, KeyedWriteError, StorageConfig},
-};
+use super::file_system;
+
+fn aws_config() -> SdkConfig {
+    block_on(async { aws_config::from_env().load().await })
```
[the remark this answers is not in the document]
Comment (chunningham) [reply src/storage/s3.rs:24]: oh nice, will remove
